bottlerocket/agents/src/userdata.rs

```diff
@@ -0,0 +1,106 @@
+use resource_agent::provider::{IntoProviderError, ProviderResult, Resources};
```

Comment:
I would prefer to use the existing lib in Bottlerocket instead of duplicating all of the code.